### ISSUE TITLE
Killing -ti container with ^C causes docker to freeze completely until restart

### ISSUE THREAD

**Issue by StrongestNumber9** [teragrep/dkr_01#17]
```sh
$ docker plugin ls
ID             NAME                 DESCRIPTION              ENABLED
60d721d8488d   teragrep/dkr_01:v1   RELP Log Driver Plugin   true

$ docker plugin disable teragrep/dkr_01:v1
Error response from daemon: plugin teragrep/dkr_01:v1 is in use

$ docker stop -t 0 kontti
<nothing happens>
```

I executed a container initially with
```sh
$ docker run --rm -ti --name kontti --log-driver=teragrep/dkr_01:v1 --log-opt relpHostname=127.0.0.1 --log-opt relpPort=1601 --log-opt relpTls=false example-container:23
docker: Error response from daemon: failed to initialize logging driver: error creating logger: Post "http://%2Frun%2Fdocker%2Fplugins%2F60d721d8488dfb7cfa58d50fa34c146eb27d515a693f134ddd90ccda4fe77ae3%2Fdkr_01.sock/LogDriver.StartLogging": net/http: request canceled (Client.Timeout exceeded while awaiting headers).
```

And again with
```
$ docker run --rm -ti --name kontti --log-driver=teragrep/dkr_01:v1 --log-opt relpHostname=127.0.0.1 --log-opt relpPort=1601 --log-opt relpTls=false example-container:23
```

Note: I did not have a relp server available at the time of launching it.

Restarting docker gives these outputs:
```sh
# The docker run command
docker: error during connect: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.41/containers/768ee65610544b30384d9f59bdaff58113ba3d5f5c2f31560b544030e52ef477/start": EOF.
ERRO[2170] error waiting for container: unexpected EOF  

# docker stop
error during connect: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.41/containers/kontti/stop?t=1": EOF
```

Restarting docker service and re-running the container allows me to show plugins, but inspecting the container stalls
```sh
$ docker plugin inspect teragrep/dkr_01:v1 | jq -r '.[0] | .Name + " -> " + .Id'
teragrep/dkr_01:v1 -> 60d721d8488dfb7cfa58d50fa34c146eb27d515a693f134ddd90ccda4fe77ae3

$ docker inspect kontti
<nothing happens>
```

---

Seems like reproduce is as follows:
- Build and install plugin
- Start relp server
- Launch a container with -ti
- Stop it abruptly with ^C
- Relaunch it
- -> No logs happening

Running kill -9 inside the container to kill it, or using docker stop $name and docker stop $name -t0 did not cause issues. Only with ^C:ing it from -ti container

**Comment by eemhu** [comment]
Seems to be reproducible, ~~might be related to the logging driver not removing the file from the internal "logs" map, however I am not sure of this yet, needs more investigation~~
(edit: Seems to be caused by a panic from the RelpConnection.Disconnect() method)

Restart prints
```sh
docker: Error response from daemon: failed to initialize logging driver: error creating logger: Post "http://%2Frun%2Fdocker%2Fplugins%2F2703087338acf8bb7c4428bbc2c560e53441480e35e515476c6a3eab80805010%2Fdkr_01.sock/LogDriver.StartLogging": net/http: request canceled (Client.Timeout exceeded while awaiting headers).
```

plugin stderr has
`FIFO Stream closed read /proc/self/fd/8: file already closed` after first CTRL-C

each attempt to run a container after CTRL-C will print "Start logging request for container <snip>" to plugin stdout

**Comment by eemhu** [comment]
Docker can be restored with `docker plugin disable PLUGIN-ID -f` and `docker plugin enable PLUGIN-ID` and starting the container again.

**Comment by eemhu** [comment]
Seems like the culprit is 
```
2023/05/08 07:43:17 http: panic serving @: Cannot disconnect, connection was not OPEN
goroutine 10 [running]:
net/http.(*conn).serve.func1()
	/usr/local/go/src/net/http/server.go:1850 +0xbf
panic({0x886960, 0x9e0f70})
	/usr/local/go/src/runtime/panic.go:890 +0x262
github.com/teragrep/rlp_05/pkg/RelpConnection.(*RelpConnection).Disconnect(0xc00006c720?)
	/go/pkg/mod/github.com/teragrep/rlp_05@v0.0.0-20230406124932-8433a16a0259/pkg/RelpConnection/RelpConnection.go:132 +0x1a7
main.(*Driver).StopLogging(0xc000076d20, {0xc0002c0180, 0x54})
	/relpapp/driver.go:272 +0xa5
```

added recover to StopLogging to force disconnect, remove from logs map and removing the lock (`d.mu.Unlock()`) seems to fix the issue. Will make a PR.

Alternatively the RLP_05 library could be modified to return an error on the Disconnect method instead of panicking.

**Comment by eemhu** [comment]
fixed in pr #18
